File: 0189-rotate-array/solution.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        vector<int> arr(n,0);
        for(int i=0;i<n;i++){
            arr[(i+k)%n]=nums[i];
        }
        for(int i=0;i<n;i++){
            nums[i]=arr[i];
        }
    }
};

File: 0189-rotate-array/solution.java
class Solution {
    public void rotate(int[] nums, int k) {
        int le = nums.length;
        int[] arr = new int[le];
        for(int i = 0; i < le; i++) {
            arr[(i + k) % le] = nums[i];
        }
        for(int i = 0; i < le; i++) {
            nums[i] = arr[i];
        }
        for(int i = 0; i < le; i++) {
            System.out.print(nums[i]);
            if (i != le - 1) {
                System.out.print(", ");
            }
        }
    }
}

